[visionOS] Update appearance of controls on spatial images

yoelhawa opened the pull request:
#### 4ef6b1a681102db45b2c19ff094ce8c19a137211
<pre>
[visionOS] Update appearance of controls on spatial images
<a href="https://bugs.webkit.org/show_bug.cgi?id=283131">https://bugs.webkit.org/show_bug.cgi?id=283131</a>
&lt;<a href="https://rdar.apple.com/138425660">rdar://138425660</a>&gt;

Reviewed by Ryosuke Niwa.

Update the design of controls for spatial images.

* LayoutTests/fast/images/spatial-image-controls-expected.html:
* LayoutTests/fast/images/spatial-image-controls.html:
Update layout test to match new design.

* Source/WebCore/html/shadow/SpatialImageControls.cpp:
(WebCore::SpatialImageControls::ensureSpatialControls):
* Source/WebCore/html/shadow/spatialImageControls.css:
(#if defined(ENABLE_SPATIAL_IMAGE_CONTROLS) &amp;&amp; ENABLE_SPATIAL_IMAGE_CONTROLS):
(div#background-tint):
(div#blur):
(div#tint):
(div#bottom-gradient):
(div#label):
(span#spatial-glyph):
Implement new design.

Canonical link: <a href="https://commits.webkit.org/286652@main">https://commits.webkit.org/286652@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/942c1791829b82d280272fe2dca0f9ea1c0340cd

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/76555 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/55590 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/29461 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/81082 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/27831 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/78672 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/64732 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/3883 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/60014 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/18121 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/79622 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/49925 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/65733 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/40339 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/47324 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/23226 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/26155 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/68447 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/23557 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/82531 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/3931 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/2582 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/68294 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/4084 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65705 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/67541 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16866 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/11509 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/9600 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/3878 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/6687 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/3901 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/7331 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/5659 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/html/shadow/SpatialImageControls.cpp:
```diff
@@ -115,10 +116,20 @@ void ensureSpatialControls(HTMLImageElement& imageElement)
             return;
 
+        StringBuilder controlStyleBuilder;
+        controlStyleBuilder.append("display: flex; flex-direction: column; justify-content: space-between; position: relative; box-sizing: border-box;"_s);
```
We should use StyledElement::setInlineStyleProperty instead of building a string to parse.

yoelhawa replied:
Good call!

Commit:
[visionOS] Update appearance of controls on spatial images
https://bugs.webkit.org/show_bug.cgi?id=283131
<rdar://138425660>

Reviewed by Ryosuke Niwa.

Update the design of controls for spatial images.

* LayoutTests/fast/images/spatial-image-controls-expected.html:
* LayoutTests/fast/images/spatial-image-controls.html:
Update layout test to match new design.

* Source/WebCore/html/shadow/SpatialImageControls.cpp:
(WebCore::SpatialImageControls::ensureSpatialControls):
* Source/WebCore/html/shadow/spatialImageControls.css:
(#if defined(ENABLE_SPATIAL_IMAGE_CONTROLS) && ENABLE_SPATIAL_IMAGE_CONTROLS):
(div#background-tint):
(div#blur):
(div#tint):
(div#bottom-gradient):
(div#label):
(span#spatial-glyph):
Implement new design.

Canonical link: https://commits.webkit.org/286652@main
